src/components/dialog/index.jsx

```diff
@@ -1,13 +1,16 @@
 import React from 'react'
 import { createPortal } from 'react-dom'
-import { useNavigate } from 'react-router-dom'
+import { useNavigate, useLocation } from 'react-router-dom'
 import FocusTrap from 'focus-trap-react'
 
 import './style.css'
 
 const Dialog = ({title, content}) => {
 	const navigate = useNavigate()
-	const onClose = () => navigate('/')
+    const location = useLocation()
+    const previousUrl = [location.state.previousLocation.pathname, location.state.previousLocation.search].join('')
+
+	const onClose = () => navigate(previousUrl)

```

Comment:
this variable feels redundant. 

```suggestion
    const {state: {previousLocation: { pathname, search }}} = useLocation();
    const onClose = () => navigate(`${pathname}${search}`)
```